src/components/Differential/ScatterPlot.jsx

```diff
@@ -1300,115 +1192,113 @@ class ScatterPlot extends React.PureComponent {
+    isTableAlreadyAccurate,
     xScale,
     yScale,
     clearHighlightedData,
-    doubleClick,
   ) {
     const self = this;
     const {
-      differentialResults,
-      differentialFeatureIdKey,
       differentialResultsUnfiltered,
+      differentialFeatureIdKey,
       filteredDifferentialTableData,
     } = self.props;
+    const { transitioning } = this.state;
+    this.removeCirclesBinsAndLabels();
 
-    d3.select('#clip-path')
-      .selectAll('path')
-      .remove();
-
-    d3.select('#clip-path')
-      .selectAll('circle')
-      .remove();
-
-    if (d3.select('#nonfiltered-elements').size() !== 0) {
-      d3.select('#nonfiltered-elements')
-        .selectAll('text')
-        .remove();
-    }
-
-    const filteredDifferentialTableDataFeatureIds = filteredDifferentialTableData.map(
+    const allFilteredDifferentialTableDataFeatureIds = filteredDifferentialTableData.map(
       tableDataElement => tableDataElement[differentialFeatureIdKey],
     );
-    const differentialResultsFeatureIds = differentialResults.map(
-      differentialResultElement =>
-        differentialResultElement[differentialFeatureIdKey],
-    );
-    const dataInViewCopy = [...dataInSelection];
-
-    const dataInView = doubleClick
-      ? differentialResultsUnfiltered
-      : dataInViewCopy;
-
-    let dataInViewPassingTableFilters =
-      dataInView.length > 0
-        ? dataInView.filter(d =>
-            filteredDifferentialTableDataFeatureIds.includes(
-              d[differentialFeatureIdKey],
-            ),
-          )
-        : [];
-
-    const dataInViewPassingTableFiltersAndMultiset =
-      dataInViewPassingTableFilters.length > 0
-        ? dataInViewPassingTableFilters.filter(divptf =>
-            differentialResultsFeatureIds.includes(
-              divptf[differentialFeatureIdKey],
-            ),
+
+    // WE MUST CALCULTATE RELEVANT AND IRRELEVANT BINS/CIRCLES
+    let irrelevantData = [];
+    let relevantData = [];
+    if (
+      !transitioning &&
+      allFilteredDifferentialTableDataFeatureIds.length ===
+        differentialResultsUnfiltered.length
+    ) {
+      // IF THE TABLE HAS THE SAME DATA LENGTH THE differentialResultsUnfiltered,
+      // THERE IS NO ZOOM NOR TABLE FILTER IN EFFECT, SO THIS SAVES US ANY CALCULATION
+      relevantData = [...differentialResultsUnfiltered];
+    } else {
+      // OTHERWISE THERE IS A ZOOM OR TABLE FILTER IN EFFECT
+      // AND WE MUST FIND THE INTERSECTION AND DIFFERENCE
+      // BETWEEN THE TABLE DATA AND (POSSIBLY ZOOMED) VIEW
+      // console.time('forEach + includes');
+      // [...dataInSelection].forEach(obj => {
+      //   if (
+      //     allFilteredDifferentialTableDataFeatureIds.includes(
+      //       obj[differentialFeatureIdKey],
+      //     )
+      //   ) {
+      //     relevantData.push(obj);
+      //   } else {
+      //     irrelevantData.push(obj);
+      //   }
+      // });
+      [...dataInSelection].forEach(obj => {
```

Comment:
This time-consuming calculation must remain, in order to separate data _within the box selection view_ into relevant (blue) and irrelevant (grey) circles and bins. If you have suggestions for a workaround or how to make it faster, let me know.